test/e2e/framework/service/jig.go

```diff
@@ -355,91 +356,18 @@ func (j *TestJig) WaitForEndpointOnNode(ctx context.Context, nodeName string) er
 
 // waitForAvailableEndpoint waits for at least 1 endpoint to be available till timeout
 func (j *TestJig) waitForAvailableEndpoint(ctx context.Context, timeout time.Duration) error {
-	ctx, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
this code is correct, a Service MUST have Endponits and EndpointSlices , both, why remove Endpoints from the assertion, this is a Services test

Reply:
this is  network e2e test by the way

Reply:
> a Service MUST have Endpoints

Currently. :smiling_imp:

> this is a Services test

It's not a _test_, it's part of the framework. None of the callers of this function care about v1.Endpoints specifically, they care about "the service proxy is aware of the endpoints", and the v1.Endpoints are completely irrelevant to that.

There are other tests in `test/e2e/network` that assert that the Endpoints controller works correctly.

Reply:
I feel safer if we keep this double check, things like that help a lot to debug flakes and troubleshoot problems with the endpoints and endpointslice controller because allows you to correlate

Reply:
There are plenty of tests in `tests/e2e/network` that assert that the Endpoints and EndpointSlice controllers give equivalent results. We don't _need_ to do it here as well.

(This PR is the first part of a plan to eventually make the Endpoints controller officially optional (to save disk and CPU), which we can only safely do if we have an e2e job that runs with the Endpoints controller disabled to confirm that everything still works in that configuration, which we can only do if we have a `[Feature:EndpointsController]` tag to wrap around all of the tests that won't work in that job. And we don't want random tests getting skipped in the no-Endpoints-controller case, when they were only trying to figure out "does kube-proxy know about the endpoints of this service?" for which looking at v1.Endpoints isn't even helpful anyway.)

Reply:
https://github.com/kubernetes/enhancements/pull/4975